### PR TITLE
Pp 13033 convert node runner pipeline

### PR DESCRIPTION
**Pull request by danworth**
### NOTES ###
The original yaml version had two `in_parallel` steps each with only one step, https://github.com/alphagov/pay-ci/blob/master/ci/pipelines/node-runner.yml#L51 and https://github.com/alphagov/pay-ci/blob/master/ci/pipelines/node-runner.yml#L71 , which AFAIK is redundant unless someone can help me see otherwise, so I have not kept that during the conversion, so there are some differences to understand, though I think its correct... 

If it helps the reviewer to see that the `build-node-18` steps are identical, since it is no longer nested in a single step `in-parrellel` here's the diff showing so:
```
➤➤ pay-ci $ diff -ys <(yq '.. | select(has("task")) | select(.task == "build-node-18") | sort_keys(.)' ci/pipelines/node-runner.yml) <(yq '.. | select(has("task")) | select(.task == "build-node-18") | sort_keys(.)' <(pkl eval ci/pkl-pipelines/pay-deploy/node-runner.pkl))
Files /dev/fd/63 and /dev/fd/62 are identical
```

### CODE REVIEW

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/node-runner.pkl

<details><summary>Diff of YAML generated from pay-deploy/node-runner.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/node-runner.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/node-runner.yml
@@ -0,0 +1,99 @@
+jobs:
+- name: update-pipeline
+  plan:
+  - get: pipeline-source
+    trigger: true
+  - task: generate-yaml-from-pkl
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: governmentdigitalservice/pay-concourse-runner
+          tag: latest
+      inputs:
+      - name: pipeline-source
+      outputs:
+      - name: pipeline-source
+      run:
+        path: ash
+        args:
+        - -c
+        - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+        dir: pipeline-source/ci/pkl-pipelines/
+  - set_pipeline: node-runner
+    file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+- name: build-and-push
+  plan:
+  - in_parallel:
+    - get: node-runner-src
+      trigger: true
+    - get: pay-ci
+  - task: generate-docker-creds-config
+    file: pay-ci/ci/tasks/generate-docker-config-file.yml
+    params:
+      USERNAME: ((docker-username))
+      PASSWORD: ((docker-access-token))
+      EMAIL: ((docker-email))
+  - task: build-node-18
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: concourse/oci-build-task
+      inputs:
+      - name: node-runner-src
+      outputs:
+      - name: image
+      run:
+        path: build
+    privileged: true
+    params:
+      CONTEXT: node-runner-src/ci/docker/node-runner
+      DOCKER_CONFIG: docker_creds
+    output_mapping:
+      image: node18-image
+  - put: node-runner-node18
+    params:
+      image: node18/image.tar
+    get_params:
+      skip_download: true
+resources:
+- name: pipeline-source
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+    - ci/pkl-pipelines/common/**
+  icon: github
+- name: node-runner-src
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/docker/node-runner/*
+  icon: github
+- name: pay-ci
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+  icon: github
+- name: node-runner-node18
+  type: registry-image
+  source:
+    repository: governmentdigitalservice/pay-node-runner
+    tag: node18
+    username: ((docker-username))
+    password: ((docker-access-token))
+  icon: docker
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/node-runner.pkl`</summary>

```diff
resources:
  resource node-runner-pipeline has been removed:
- icon: github
- name: node-runner-pipeline
- source:
-   branch: master
-   paths:
-   - ci/pipelines/node-runner.yml
-   uri: https://github.com/alphagov/pay-ci
- type: git
  
  resource node-runner-src has changed:
  icon: github
  name: node-runner-src
  source:
    branch: master
+   password: ((github-access-token))
    paths:
    - ci/docker/node-runner/*
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pay-ci has changed:
  icon: github
  name: pay-ci
  source:
    branch: master
+   password: ((github-access-token))
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pipeline-source has been added:
+ icon: github
+ name: pipeline-source
+ source:
+   branch: master
+   password: ((github-access-token))
+   paths:
+   - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+   - ci/pkl-pipelines/common/**
+   uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
+ type: git
  
jobs:
  job build-and-push has changed:
  name: build-and-push
  plan:
  - in_parallel:
      steps:
      - get: node-runner-src
        trigger: true
      - get: pay-ci
  - file: pay-ci/ci/tasks/generate-docker-config-file.yml
    params:
      EMAIL: ((docker-email))
      PASSWORD: ((docker-access-token))
      USERNAME: ((docker-username))
    task: generate-docker-creds-config
- - in_parallel:
-     steps:
-     - config:
-         image_resource:
-           name: ""
-           source:
-             repository: concourse/oci-build-task
-           type: registry-image
-         inputs:
-         - name: node-runner-src
-         outputs:
-         - name: image
-         platform: linux
-         run:
-           path: build
-       output_mapping:
-         image: node18-image
-       params:
-         CONTEXT: node-runner-src/ci/docker/node-runner
-         DOCKER_CONFIG: docker_creds
-       privileged: true
-       task: build-node-18
- - in_parallel:
-     steps:
-     - do: null
-     - get_params:
-         skip_download: true
-       params:
-         image: node18-image/image.tar
-       put: node-runner-node18
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: concourse/oci-build-task
+       type: registry-image
+     inputs:
+     - name: node-runner-src
+     outputs:
+     - name: image
+     platform: linux
+     run:
+       path: build
+   output_mapping:
+     image: node18-image
+   params:
+     CONTEXT: node-runner-src/ci/docker/node-runner
+     DOCKER_CONFIG: docker_creds
+   privileged: true
+   task: build-node-18
+ - get_params:
+     skip_download: true
+   params:
+     image: node18/image.tar
+   put: node-runner-node18
  
  job update-node-runner-pipeline has been removed:
- name: update-node-runner-pipeline
- plan:
- - get: node-runner-pipeline
-   trigger: true
- - file: node-runner-pipeline/ci/pipelines/node-runner.yml
-   set_pipeline: node-runner
  
  job update-pipeline has been added:
+ name: update-pipeline
+ plan:
+ - get: pipeline-source
+   trigger: true
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: governmentdigitalservice/pay-concourse-runner
+         tag: latest
+       type: registry-image
+     inputs:
+     - name: pipeline-source
+     outputs:
+     - name: pipeline-source
+     platform: linux
+     run:
+       args:
+       - -c
+       - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+       dir: pipeline-source/ci/pkl-pipelines/
+       path: ash
+   task: generate-yaml-from-pkl
+ - file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+   set_pipeline: node-runner
  
pipeline name: node-runner

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/node-runner.pkl

<details><summary>Diff of YAML generated from pay-deploy/node-runner.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/node-runner.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/node-runner.yml
@@ -0,0 +1,99 @@
+jobs:
+- name: update-pipeline
+  plan:
+  - get: pipeline-source
+    trigger: true
+  - task: generate-yaml-from-pkl
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: governmentdigitalservice/pay-concourse-runner
+          tag: latest
+      inputs:
+      - name: pipeline-source
+      outputs:
+      - name: pipeline-source
+      run:
+        path: ash
+        args:
+        - -c
+        - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+        dir: pipeline-source/ci/pkl-pipelines/
+  - set_pipeline: node-runner
+    file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+- name: build-and-push
+  plan:
+  - in_parallel:
+    - get: node-runner-src
+      trigger: true
+    - get: pay-ci
+  - task: generate-docker-creds-config
+    file: pay-ci/ci/tasks/generate-docker-config-file.yml
+    params:
+      USERNAME: ((docker-username))
+      PASSWORD: ((docker-access-token))
+      EMAIL: ((docker-email))
+  - task: build-node-18
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: concourse/oci-build-task
+      inputs:
+      - name: node-runner-src
+      outputs:
+      - name: image
+      run:
+        path: build
+    privileged: true
+    params:
+      CONTEXT: node-runner-src/ci/docker/node-runner
+      DOCKER_CONFIG: docker_creds
+    output_mapping:
+      image: node18-image
+  - put: node-runner-node18
+    params:
+      image: node18-image/image.tar
+    get_params:
+      skip_download: true
+resources:
+- name: pipeline-source
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+    - ci/pkl-pipelines/common/**
+  icon: github
+- name: node-runner-src
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/docker/node-runner/*
+  icon: github
+- name: pay-ci
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+  icon: github
+- name: node-runner-node18
+  type: registry-image
+  source:
+    repository: governmentdigitalservice/pay-node-runner
+    tag: node18
+    username: ((docker-username))
+    password: ((docker-access-token))
+  icon: docker
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/node-runner.pkl`</summary>

```diff
resources:
  resource node-runner-pipeline has been removed:
- icon: github
- name: node-runner-pipeline
- source:
-   branch: master
-   paths:
-   - ci/pipelines/node-runner.yml
-   uri: https://github.com/alphagov/pay-ci
- type: git
  
  resource node-runner-src has changed:
  icon: github
  name: node-runner-src
  source:
    branch: master
+   password: ((github-access-token))
    paths:
    - ci/docker/node-runner/*
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pay-ci has changed:
  icon: github
  name: pay-ci
  source:
    branch: master
+   password: ((github-access-token))
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pipeline-source has been added:
+ icon: github
+ name: pipeline-source
+ source:
+   branch: master
+   password: ((github-access-token))
+   paths:
+   - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+   - ci/pkl-pipelines/common/**
+   uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
+ type: git
  
jobs:
  job build-and-push has changed:
  name: build-and-push
  plan:
  - in_parallel:
      steps:
      - get: node-runner-src
        trigger: true
      - get: pay-ci
  - file: pay-ci/ci/tasks/generate-docker-config-file.yml
    params:
      EMAIL: ((docker-email))
      PASSWORD: ((docker-access-token))
      USERNAME: ((docker-username))
    task: generate-docker-creds-config
- - in_parallel:
-     steps:
-     - config:
-         image_resource:
-           name: ""
-           source:
-             repository: concourse/oci-build-task
-           type: registry-image
-         inputs:
-         - name: node-runner-src
-         outputs:
-         - name: image
-         platform: linux
-         run:
-           path: build
-       output_mapping:
-         image: node18-image
-       params:
-         CONTEXT: node-runner-src/ci/docker/node-runner
-         DOCKER_CONFIG: docker_creds
-       privileged: true
-       task: build-node-18
- - in_parallel:
-     steps:
-     - do: null
-     - get_params:
-         skip_download: true
-       params:
-         image: node18-image/image.tar
-       put: node-runner-node18
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: concourse/oci-build-task
+       type: registry-image
+     inputs:
+     - name: node-runner-src
+     outputs:
+     - name: image
+     platform: linux
+     run:
+       path: build
+   output_mapping:
+     image: node18-image
+   params:
+     CONTEXT: node-runner-src/ci/docker/node-runner
+     DOCKER_CONFIG: docker_creds
+   privileged: true
+   task: build-node-18
+ - get_params:
+     skip_download: true
+   params:
+     image: node18-image/image.tar
+   put: node-runner-node18
  
  job update-node-runner-pipeline has been removed:
- name: update-node-runner-pipeline
- plan:
- - get: node-runner-pipeline
-   trigger: true
- - file: node-runner-pipeline/ci/pipelines/node-runner.yml
-   set_pipeline: node-runner
  
  job update-pipeline has been added:
+ name: update-pipeline
+ plan:
+ - get: pipeline-source
+   trigger: true
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: governmentdigitalservice/pay-concourse-runner
+         tag: latest
+       type: registry-image
+     inputs:
+     - name: pipeline-source
+     outputs:
+     - name: pipeline-source
+     platform: linux
+     run:
+       args:
+       - -c
+       - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+       dir: pipeline-source/ci/pkl-pipelines/
+       path: ash
+   task: generate-yaml-from-pkl
+ - file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+   set_pipeline: node-runner
  
pipeline name: node-runner

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

File: ci/pkl-pipelines/pay-deploy/node-runner.pkl [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    name = "node-runner-src"
+    repoName = "pay-ci"
+    source {
+      branch = "master"
+      paths = new {
+        "ci/docker/node-runner/*"
+      }
+    }
+  }
+  new PayResources.PayGitHubResource {
```
Comment (jfharden) [inline ci/pkl-pipelines/pay-deploy/node-runner.pkl:18]: Would be good to use the shared resource which defines this https://github.com/alphagov/pay-ci/blob/master/ci/pkl-pipelines/common/shared_resources.pkl#L14

Comment (danworth) [reply ci/pkl-pipelines/pay-deploy/node-runner.pkl:18]: ah good one 👍 , I think the challenge here is finding the pre-made shared resources sometimes, I had a cursory look but nothing jumped out.

File: ci/pkl-pipelines/pay-deploy/node-runner.pkl [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  new {
+    name = "build-and-push"
+    plan {
+      new InParallelStep {
+        in_parallel = new Listing<Step> {
+          new GetStep {get = "node-runner-src" trigger = true}
+          new GetStep {get = "pay-ci"}
+        }
+      }
+      new TaskStep {
```
Comment (jfharden) [inline ci/pkl-pipelines/pay-deploy/node-runner.pkl:45]: We've got this in shared resources too, called generateDockerCredsConfigStep

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/node-runner.pkl

<details><summary>Diff of YAML generated from pay-deploy/node-runner.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/node-runner.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/node-runner.yml
@@ -0,0 +1,99 @@
+jobs:
+- name: update-pipeline
+  plan:
+  - get: pipeline-source
+    trigger: true
+  - task: generate-yaml-from-pkl
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: governmentdigitalservice/pay-concourse-runner
+          tag: latest
+      inputs:
+      - name: pipeline-source
+      outputs:
+      - name: pipeline-source
+      run:
+        path: ash
+        args:
+        - -c
+        - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+        dir: pipeline-source/ci/pkl-pipelines/
+  - set_pipeline: node-runner
+    file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+- name: build-and-push
+  plan:
+  - in_parallel:
+    - get: node-runner-src
+      trigger: true
+    - get: pay-ci
+  - task: generate-docker-creds-config
+    file: pay-ci/ci/tasks/generate-docker-config-file.yml
+    params:
+      USERNAME: ((docker-username))
+      PASSWORD: ((docker-access-token))
+      EMAIL: ((docker-email))
+  - task: build-node-18
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: concourse/oci-build-task
+      inputs:
+      - name: node-runner-src
+      outputs:
+      - name: image
+      run:
+        path: build
+    privileged: true
+    params:
+      CONTEXT: node-runner-src/ci/docker/node-runner
+      DOCKER_CONFIG: docker_creds
+    output_mapping:
+      image: node18-image
+  - put: node-runner-node18
+    params:
+      image: node18-image/image.tar
+    get_params:
+      skip_download: true
+resources:
+- name: pipeline-source
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+    - ci/pkl-pipelines/common/**
+  icon: github
+- name: node-runner-src
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+    paths:
+    - ci/docker/node-runner/*
+  icon: github
+- name: pay-ci
+  type: git
+  source:
+    uri: https://github.com/alphagov/pay-ci
+    branch: master
+    username: alphagov-pay-ci-concourse
+    password: ((github-access-token))
+  icon: github
+- name: node-runner-node18
+  type: registry-image
+  source:
+    repository: governmentdigitalservice/pay-node-runner
+    tag: node18
+    username: ((docker-username))
+    password: ((docker-access-token))
+  icon: docker
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/node-runner.pkl`</summary>

```diff
resources:
  resource node-runner-pipeline has been removed:
- icon: github
- name: node-runner-pipeline
- source:
-   branch: master
-   paths:
-   - ci/pipelines/node-runner.yml
-   uri: https://github.com/alphagov/pay-ci
- type: git
  
  resource node-runner-src has changed:
  icon: github
  name: node-runner-src
  source:
    branch: master
+   password: ((github-access-token))
    paths:
    - ci/docker/node-runner/*
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pay-ci has changed:
  icon: github
  name: pay-ci
  source:
    branch: master
+   password: ((github-access-token))
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git
  
  resource pipeline-source has been added:
+ icon: github
+ name: pipeline-source
+ source:
+   branch: master
+   password: ((github-access-token))
+   paths:
+   - ci/pkl-pipelines/pay-deploy/node-runner.pkl
+   - ci/pkl-pipelines/common/**
+   uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
+ type: git
  
jobs:
  job build-and-push has changed:
  name: build-and-push
  plan:
  - in_parallel:
      steps:
      - get: node-runner-src
        trigger: true
      - get: pay-ci
  - file: pay-ci/ci/tasks/generate-docker-config-file.yml
    params:
      EMAIL: ((docker-email))
      PASSWORD: ((docker-access-token))
      USERNAME: ((docker-username))
    task: generate-docker-creds-config
- - in_parallel:
-     steps:
-     - config:
-         image_resource:
-           name: ""
-           source:
-             repository: concourse/oci-build-task
-           type: registry-image
-         inputs:
-         - name: node-runner-src
-         outputs:
-         - name: image
-         platform: linux
-         run:
-           path: build
-       output_mapping:
-         image: node18-image
-       params:
-         CONTEXT: node-runner-src/ci/docker/node-runner
-         DOCKER_CONFIG: docker_creds
-       privileged: true
-       task: build-node-18
- - in_parallel:
-     steps:
-     - do: null
-     - get_params:
-         skip_download: true
-       params:
-         image: node18-image/image.tar
-       put: node-runner-node18
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: concourse/oci-build-task
+       type: registry-image
+     inputs:
+     - name: node-runner-src
+     outputs:
+     - name: image
+     platform: linux
+     run:
+       path: build
+   output_mapping:
+     image: node18-image
+   params:
+     CONTEXT: node-runner-src/ci/docker/node-runner
+     DOCKER_CONFIG: docker_creds
+   privileged: true
+   task: build-node-18
+ - get_params:
+     skip_download: true
+   params:
+     image: node18-image/image.tar
+   put: node-runner-node18
  
  job update-node-runner-pipeline has been removed:
- name: update-node-runner-pipeline
- plan:
- - get: node-runner-pipeline
-   trigger: true
- - file: node-runner-pipeline/ci/pipelines/node-runner.yml
-   set_pipeline: node-runner
  
  job update-pipeline has been added:
+ name: update-pipeline
+ plan:
+ - get: pipeline-source
+   trigger: true
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: governmentdigitalservice/pay-concourse-runner
+         tag: latest
+       type: registry-image
+     inputs:
+     - name: pipeline-source
+     outputs:
+     - name: pipeline-source
+     platform: linux
+     run:
+       args:
+       - -c
+       - mkdir -p 'pay-deploy' && pkl eval 'pay-deploy/node-runner.pkl' > 'pay-deploy/node-runner.yaml'
+       dir: pipeline-source/ci/pkl-pipelines/
+       path: ash
+   task: generate-yaml-from-pkl
+ - file: pipeline-source/ci/pkl-pipelines/pay-deploy/node-runner.yaml
+   set_pipeline: node-runner
  
pipeline name: node-runner

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Review by jfharden — APPROVED** [review]

**Pull request merged**